### PR TITLE
Fix Kafka Monitoring

### LINKED ISSUE #229: Kafka Monitoring is broken
We are currently unable to view kafka metrics in grafana. Prometheus shows that it is completely unable to scrape kafka metrics.

### PR DESCRIPTION
**Pull request by anishasthana**
This PR will fix our Kafka monitoring and close #229

### CODE REVIEW

File: prometheus/operator/base/kafka-podmonitors.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      regex: (.*)
+      targetLabel: node_name
+      replacement: $1
+      action: replace
+    - sourceLabels: [__meta_kubernetes_pod_host_ip]
+      separator: ;
+      regex: (.*)
+      targetLabel: node_ip
+      replacement: $1
+      action: replace
```
Comment (vpavlin) [inline prometheus/operator/base/kafka-podmonitors.yaml:102]: missing new line:)

File: prometheus/operator/base/kafka-podmonitors.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  labels:
+    app: strimzi
+    team: opendatahub
+spec:
+  selector:
+    matchExpressions:
+      - key: "strimzi.io/kind"
+        operator: In
+        values: ["Kafka", "KafkaConnect"]
+  namespaceSelector:
```
Comment (vpavlin) [inline prometheus/operator/base/kafka-podmonitors.yaml:68]: Does this mean that it will only work in `opendatahub` namespace?

**Comment by crobby** [comment]
@anishasthana Might be an actual problem:  
`time="2020-11-05T14:52:38Z" level=error msg="Error evaluating kustomization manifest for prometheus-operator: accumulating resources: recursed accumulation of path 'base': accumulating resources: accumulating resources from 'kafka-servicemonitor.yaml': open /tmp/opendatahub/opendatahub/kustomize/prometheus-operator/base/kafka-servicemonitor.yaml: no such file or directory" `

File: prometheus/operator/base/kafka-podmonitors.yaml (hunk 2 above)
Comment (anishasthana) [reply prometheus/operator/base/kafka-podmonitors.yaml]: Good catch

File: prometheus/operator/base/kafka-podmonitors.yaml (hunk 1 above)
Comment (anishasthana) [reply prometheus/operator/base/kafka-podmonitors.yaml]: Fixed

**Comment by anishasthana** [comment]
@vpavlin @crobby tests passed :-)

**Review by crobby — APPROVED** [review]
Looks solid to me.

File: prometheus/operator/base/kafka-podmonitors.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (anishasthana) [reply prometheus/operator/base/kafka-podmonitors.yaml]: Are you looking at the latest diff?

**Pull request merged**